ui/src/api_tests/api.spec.ts

```diff
@@ -353,6 +375,76 @@ describe('main', () => {
       expect(getPostRes.post.name).toBe(name);
       expect(getPostRes.post.community_local).toBe(true);
       expect(getPostRes.post.creator_local).toBe(false);
+      expect(getPostRes.post.stickied).toBe(true);
+
+      let lockPostForm: LockPostForm = {
+        edit_id: 2,
+        locked: true,
+        auth: lemmyAlphaAuth,
+      };
+
+      let lockedRes: PostResponse = await fetch(
+        `${lemmyAlphaApiUrl}/post/lock`,
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: wrapper(lockPostForm),
+        }
+      ).then(d => d.json());
+
+      expect(lockedRes.post.name).toBe(name);
+      expect(lockedRes.post.locked).toBe(true);
+
+      // Fetch from B to make sure its locked
+      getPostRes = await fetch(getPostUrl, {
+        method: 'GET',
+      }).then(d => d.json());
+      expect(getPostRes.post.locked).toBe(true);
+
+      // Create a test comment on a locked post, it should be undefined
+      // since it shouldn't get created.
+      let content = 'A rejected comment on a locked post';
+      let commentForm: CommentForm = {
+        content,
+        post_id: 2,
+        auth: lemmyAlphaAuth,
+      };
+
+      let createResponse: CommentResponse = await fetch(
+        `${lemmyAlphaApiUrl}/comment`,
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: wrapper(commentForm),
+        }
+      ).then(d => d.json());
+
+      expect(createResponse.comment).toBeUndefined();
```

Comment:
Does this not send a proper error message?

Reply:
Good call, looks like it does return `{ error: 'locked' }`. I'll add this instead.